crates/runtime/src/lib.rs

```diff
@@ -838,10 +838,26 @@ impl Runtime {
 
         tracing::info!("Initialized query results cache; {cache_provider}");
 
+        let shared_cache_provider = Arc::new(cache_provider);
+        let cache_copy = Arc::clone(&shared_cache_provider);
```

Comment:
This is poor naming. It's not a copy of the cache, but an additional pointer to it.

Reply:
This is now resolved. Thank you